R/diagnostic_test.R

```diff
@@ -0,0 +1,198 @@
+#' Diagnostic testing
+#'
+#' @description
+#' Run a set of diagnostic tests to check for common issues found when setting
+#' up R on a DfE
+#' system. This includes:
+#'   - Checking for proxy settings in the Git configuration
+#'   - Checking for correct download method used by renv (curl)
+#'
+#' @inheritParams check_proxy_settings
+#'
+#' @return List object of detected parameters
+#' @export
+#'
+#' @examples
+#' diagnostic_test()
+diagnostic_test <- function(
+    clean = FALSE,
+    verbose = FALSE) {
+  results <- c(
+    check_proxy_settings(clean = clean, verbose = verbose),
+    check_github_pat(clean = clean, verbose = verbose),
+    check_renv_download_method(clean = clean, verbose = verbose)
+  )
+  return(results)
+}
+
+#' Check proxy settings
+#'
+#' @description
+#' This script checks for "bad" proxy settings. Prior to the pandemic, analysts
+#' in the DfE would need to add some proxy settings to their GitHub config.
+#' These settings now prevent Git from connecting to remote archives on GitHub
+#' and Azure DevOps if present, so this script identifies and (if clean=TRUE is
+#' set) removes them.
+#'
+#' @param proxy_setting_names Vector of proxy parameters to check for. Default:
+#' c("http.proxy", "https.proxy")
+#' @param clean Attempt to clean settings
+#' @param verbose Run in verbose mode
+#'
+#' @return List of problem proxy settings
+#' @export
+#'
+#' @examples
+#' check_proxy_settings()
+check_proxy_settings <- function(
```

Comment:
Doesn't seem to be working, I've set http.proxy in my Git config 
![image](https://github.com/user-attachments/assets/ec612dee-320d-437b-84d2-2391fea04690)

But running the function even after restarting R Studio doesn't seem to pick it up? Am I doing something wrong here?
![image](https://github.com/user-attachments/assets/11f1e89f-1ab2-49eb-9360-901f86107938)